mw-webapp/src/component/sidebar/SidebarContent/SidebarContent.tsx

```diff
@@ -8,14 +8,38 @@ import {
 import {Cross2Icon} from "@radix-ui/react-icons";
 import styles from "src/component/sidebar/SidebarContent/SidebarContent.module.scss";
 
+/**
+ * SidebarContent props
+ */
+interface SidebarContentProps extends PropsWithChildren {
+
+  /**
+   * Callback triggered on SidebarContent click
+   */
+  onClick: () => void;
+}
+
 /**
  * A container for the content to be displayed within a Sidebar.
  */
-export const SidebarContent = (props: PropsWithChildren) => {
+export const SidebarContent = (props: SidebarContentProps) => {
+
+  /**
+   * Handle onClick event
+   */
+  const onClickHandler = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (event.target instanceof HTMLAnchorElement) {
+      props.onClick();
+    }
```

Comment:
interesting desion. It's better to handle  click only on link natively instead of checking event target, but if it is hard you can skip it for now. Delegation  is a good pattern for js-projects usually but not for react-projects

Reply:
to be honest, I can't imagine how to implement it)))